query/src/sql/planner/semantic/type_check.rs

```diff
@@ -933,4 +940,284 @@ impl<'a> TypeChecker<'a> {
             tuple_func.return_type(),
         ))
     }
+
+    async fn resolve_udf(
+        &mut self,
+        func_name: &str,
+        arguments: &[Expr<'_>],
+    ) -> Result<(Scalar, DataTypeImpl)> {
+        let udf = self
+            .ctx
+            .get_user_manager()
+            .get_udf(self.ctx.get_tenant().as_str(), func_name)
+            .await;
+        return if let Ok(udf) = udf {
+            let parameters = udf.parameters;
+            if parameters.len() != arguments.len() {
+                return Err(ErrorCode::SyntaxException(format!(
+                    "Require {} parameters, but got: {}",
+                    parameters.len(),
+                    arguments.len()
+                )));
+            }
+            let backtrace = Backtrace::new();
+            let sql_tokens = tokenize_sql(udf.definition.as_str())?;
+            let expr = parse_expr(&sql_tokens, &backtrace)?;
+            let mut args_map = HashMap::new();
+            arguments.iter().enumerate().for_each(|(idx, argument)| {
+                if let Some(parameter) = parameters.get(idx) {
+                    args_map.insert(parameter, (*argument).clone());
+                }
+            });
+            let udf_expr = self.clone_expr_with_replacement(&expr, &|nest_expr| {
+                if let Expr::ColumnRef { column, .. } = nest_expr {
+                    if let Some(arg) = args_map.get(&column.name) {
+                        return Ok(Some(arg.clone()));
+                    }
+                }
+                Ok(None)
+            })?;
+            self.resolve(&udf_expr, None).await
+        } else {
+            Err(ErrorCode::SemanticError(
+                "No function matches the given name.",
+            ))
+        };
+    }
+
+    fn clone_expr_with_replacement<F>(
```

Comment:
It's better to do this with a unified visitor, but we can open another issue for this.

Reply:
Ok